New Topic: Open Source

Dusch4593 opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

The following major work was done in this PR:

- Added "Open Source" to the `tags.md` file.
- Created a brand new "Open Source" topic file. (General concept entry removed)
- Created a new Open Source concept called "Open Source Software".
  - Includes an extensive table listing different frameworks, languages, etc., that are OSS.
- Migrated a lot of General concept entries into the Open Source topic**.
  - This is both to buff up the new Open Source topic and thin out/refocus the General topic.
  - **Also migrated entries for "Ruby on Rails" and "React Native" into Ruby and React topics, respectively.
- Removed the General: Cloud Computing entry in anticipation of #930 being merged before this one.
  - This is to prevent potential merge conflicts.
- Revised as many links via SourceGraph that would have broken otherwise. (Let me know if you find any more 😄)
- Some moderate word-smithing.

**Note**: This is a _huge pull request_, so let's conduct at least four (4) reviews.

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding new entries
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski commented:
@Dusch4593, I deleted the General: Cloud Computing entry in my PR. Is this going to cause a problem?

SSwiniarski commented on content/general/concepts/back-end/back-end.md:
```diff
@@ -33,13 +33,13 @@ at a restaurant (on the "front-end):
 ## Back-End Tools
 
-Back-End Developers use a range of technology and software, many of which fall into three categories: databases, programming languages, and frameworks.
+Back-End Developers use a range of technologies and softwares, many of which fall into three categories: databases, programming languages, and frameworks.
```
```suggestion
Back-End Developers use a range of technologies and software, many of which fall into three categories: databases, programming languages, and frameworks.
```

SSwiniarski commented on content/general/concepts/framework/framework.md:
```diff
@@ -14,7 +14,7 @@ CatalogContent:
 <link rel="canonical" href="https://www.codecademy.com/resources/blog/what-is-a-framework/" />
 
-Frameworks are softwares that provide reusable boilerplate code and functionality for building applications. They are commonly associated with a specific programming language and are suited to different types of tasks.
+**Frameworks** are softwares that provide reusable boilerplate code and functionality for building applications. They are commonly associated with a specific programming language and are suited to different types of tasks.
```
```suggestion
**Frameworks** are software that provide reusable boilerplate code and functionality for building applications. They are commonly associated with a specific programming language and are suited to different types of tasks.
```

Dusch4593 commented:
> @Dusch4593, I deleted the General: Cloud Computing entry in my PR. Is this going to cause a problem?

I don't think it should as long as our separate PRs didn't make separate/different changes in the same file(s).

SSwiniarski commented on content/open-source/concepts/open-source-software/open-source-software.md:
```diff
@@ -0,0 +1,42 @@
+  - 'paths/code-foundations'
+---
+
+**Open source software (OSS)** is source code that, through licensing, available for view, modification, and distribution. Although anyone in the public can contribute to OSS, it is usually maintained by a community of volunteer software developers who collaborate on websites like [GitHub](https://www.codecademy.com/resources/docs/general/github).
```
```suggestion
**Open source software (OSS)** is source code that, through licensing, is available for view, modification, and distribution. Although anyone in the public can contribute to OSS, it is usually maintained by a community of volunteer software developers who collaborate on websites like [GitHub](https://www.codecademy.com/resources/docs/general/github).
```

SSwiniarski reviewed (changes requested):
@Dusch4593 I reviewed this and made come comments.

SSwiniarski commented:
@Dusch4593, it turns out that we're both making changes to the GraphQL entry.

Dusch4593 commented:
> @Dusch4593, it turns out that we're both making changes to the GraphQL entry.

True. I wanted to make some changes as well as move the GraphQL entry inside the open source topic. To prevent conflicts, could we do the following:
 - In this PR, I'll incorporate the changes to the entry you made in your PR.
- After I let you know I made those edits, could you remove the GraphQL entry from your PR altogether?

SSwiniarski commented:
> Collaborator

@Dusch4593 Sounds good.

SSwiniarski commented:
> > @Dusch4593, it turns out that we're both making changes to the GraphQL entry.
> 
> True. I wanted to make some changes as well as move the GraphQL entry inside the open source topic. To prevent conflicts, could we do the following:
> 
> * In this PR, I'll incorporate the changes to the entry you made in your PR.
> * After I let you know I made those edits, could you remove the GraphQL entry from your PR altogether?

@Dusch4593 Sounds good.

Dusch4593 commented:
@SSwiniarski Ok, the changes were made 😄

Dusch4593 commented on content/general/concepts/code-editor/code-editor.md:
```diff
@@ -25,6 +25,5 @@ There are numerous code editors available, such as:
 - [Atom](https://atom.io/)
 - [Notepad++](https://notepad-plus-plus.org/)
-- [Vim](https://www.vim.org/)
 
```
I think I removed this because Vim is technically a _text_ editor, not a code editor.

Dusch4593 replied:
We have a link to Visual Studio (the IDE) in the IDE entry. But what if we included in this entry a link to its code editor, Visual Studio Code?

https://code.visualstudio.com/

SSwiniarski replied:
@Dusch4593, sounds good.

Dusch4593 commented:
@SSwiniarski Finished implementing your feedback. I had just one follow-up question to one of your comments/suggestions.

HishamT commented on content/open-source/concepts/open-source-software/open-source-software.md:
```diff
@@ -0,0 +1,42 @@
+  - 'paths/code-foundations'
+---
+
+**Open source software (OSS)** is source code that, through licensing, is available for view, modification, and distribution. Although anyone in the public can contribute to OSS, it is usually maintained by a community of volunteer software developers who collaborate on websites like [GitHub](https://www.codecademy.com/resources/docs/general/github).
```
Although anyone ~~in the public~~ can contribute to OSS, it is usually maintained by a community of volunteer software developers who collaborate on websites like [GitHub](https://www.codecademy.com/resources/docs/general/github).

HishamT commented on content/open-source/concepts/open-source-software/open-source-software.md:
```diff
@@ -0,0 +1,42 @@
+
+**Open source software (OSS)** is source code that, through licensing, is available for view, modification, and distribution. Although anyone in the public can contribute to OSS, it is usually maintained by a community of volunteer software developers who collaborate on websites like [GitHub](https://www.codecademy.com/resources/docs/general/github).
+
+OSS is also thought to be more trustworthy. Since all the code is available to the public, any unwanted or nefarious behavior or activity will be quickly exposed by the development community surrounding the project.
```
Since all the code is ~~available to the public~~ publicly available, any unwanted or nefarious behavior or activity will be quickly exposed by the development community ~~surrounding~~ maintaining the project.

HishamT commented on content/open-source/concepts/webpack/webpack.md:
```diff
@@ -0,0 +1,21 @@
+
+**Webpack** is a free and open-source [JavaScript module](https://www.codecademy.com/resources/docs/javascript/modules) that can also be applied to [HTML](https://www.codecademy.com/resources/docs/html) and [CSS](https://www.codecademy.com/resources/docs/css). Webpack can be run on the [Linux](https://www.codecademy.com/resources/docs/open-source/linux), macOS, and Windows operating systems.
+
+As a module bundler, Webpack can be used with `webpack.config.js` to generate dependency graphs, and define rules for tasks performed on a site. Primarily it is used by developers to load, transform, and package modules for in-browser use.
```
As a module bundler, Webpack can be used with `webpack.config.js` to generate dependency graphs and define rules for tasks performed on a site. Primarily it is used by developers to load, transform, and package modules for in-browser use.

HishamT replied:
Removed comma.

HishamT commented on content/open-source/open-source.md:
```diff
@@ -0,0 +1,11 @@
+---
```
Should this stay as "open source" and not "open-source"?

Dusch4593 replied:
While both are totally acceptable, I think the hyphen `-` should be used whenever it is used as an adjective preceding a noun (i.e., "This is an _open-source_ project", "_Open-source_ software"). In other cases I'd omit the hyphen (i.e., "_Open source_ is ...")

HishamT commented on content/react/concepts/react-native/react-native.md:
```diff
@@ -16,9 +17,9 @@ CatalogContent:
 <link rel="canonical" href="https://www.codecademy.com/resources/blog/what-is-react-native/" />
 
-[React Native](https://reactnative.dev/) is an open-source mobile development framework that works across different operating systems like iOS and Android. It was released in 2015 by Facebook (now Meta) and can be used on many smart devices, such as phones, personal computers, and TVs, to provide a consistent interface and user experience.
+[**React Native**](https://reactnative.dev/) is an [open-source](https://www.codecademy.com/resources/docs/open-source) mobile development [framework](https://www.codecademy.com/resources/docs/general/framework) that works across different operating systems like iOS and Android. It was released in 2015 by Facebook (now Meta) and can be used on many smart devices, such as phones, personal computer ...
```
that works across different operating systems ~~like~~ such as iOS and Android

HishamT commented on content/react/concepts/react-native/react-native.md:
```diff
@@ -16,9 +17,9 @@ CatalogContent:
+[**React Native**](https://reactnative.dev/) is an [open-source](https://www.codecademy.com/resources/docs/open-source) mobile development [framework](https://www.codecademy.com/resources/docs/general/framework) that works across different operating systems like iOS and Android. It was released in 2015 by Facebook (now Meta) and can be used on many smart devices, such as phones, personal computer ...
 
-Traditional mobile app development required multiple codebases for each operating system (iOS, Android, AppleTV, etc.) along with their associated programming languages ([Swift](https://www.codecademy.com/resources/docs/swift), [Java](https://www.codecademy.com/resources/docs/java), etc.). However, React Native provides a single codebase written in [React](https://www.codecademy.com/resources/doc ...
+Traditional mobile app development required multiple codebases for each operating system (iOS, Android, AppleTV, etc.) along with their associated programming languages such as [Swift](https://www.codecademy.com/resources/docs/swift) and [Java](https://www.codecademy.com/resources/docs/java). However, React Native provides a single codebase written in [React](https://www.codecademy.com/resources/ ...
```
Google recommends that Android apps should now be written in Kotlin as opposed to Java.

HishamT commented on content/react/react.md:
```diff
@@ -9,8 +9,10 @@ CatalogContent:
 <link rel="canonical" href="https://www.codecademy.com/resources/blog/what-is-react/" />
 
-[React](https://reactjs.org/) is a free and open-source front-end JavaScript library for building user interfaces or webpage components. It was released in 2013 and is currently maintained by Facebook along with a community of developers and companies.
+[**React**](https://reactjs.org/) is a free and [open source](https://www.codecademy.com/resources/docs/open-source) [JavaScript](https://www.codecademy.com/resources/docs/javascript) library for building user interfaces or [front-end](https://www.codecademy.com/resources/docs/general/front-end) webpage components such as forms along wiht fields like text areas and buttons.
```
~~[open source]~~ [open-source]

...webpage components such as forms along ~~wiht~~ with fields ~~like~~ such as areas and buttons.

HishamT reviewed (changes requested):
Looks good, but some changes are needed.

SSwiniarski commented:
> > > @Dusch4593, it turns out that we're both making changes to the GraphQL entry.
> > 
> > 
> > True. I wanted to make some changes as well as move the GraphQL entry inside the open source topic. To prevent conflicts, could we do the following:
> > 
> > * In this PR, I'll incorporate the changes to the entry you made in your PR.
> > * After I let you know I made those edits, could you remove the GraphQL entry from your PR altogether?
> 
> @Dusch4593 Sounds good.

@Dusch4593 I removed my edits to the GraphQL entry in my PR so we should avoid a conflict. The only changes I was making was bolding the first occurrence of "GraphQL" and linking the reference to "cloud services" to the new Cloud Computing topic.

Dusch4593 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,163 @@
+
+## JS and Node.js
+
+JS has existed since 1995 and has since taken over as the dominant language for web development. For much of its life, JS was used mainly for client-side scripting inside [`<script>`](https://www.codecademy.com/resources/docs/html/elements/script) elements executing in web browsers. This limitation meant that developers were often working in many different languages and frameworks between the [fr ...
```
Looks like this entry already links to General: Back-End towards the top of the entry (on line # 17).

Dusch4593 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,163 @@
+
+JS has existed since 1995 and has since taken over as the dominant language for web development. For much of its life, JS was used mainly for client-side scripting inside [`<script>`](https://www.codecademy.com/resources/docs/html/elements/script) elements executing in web browsers. This limitation meant that developers were often working in many different languages and frameworks between the [fr ...
+
+Although there were other projects to bring JS to server-side applications, the functionality took off with the launch of Node.js in 2009. Node allows developers to write JS code that runs directly in a computer process itself instead of in a browser. Node can, therefore, be used to write server-side applications with access to the operating system, file system, and everything else required to bu ...
```
It looks like much of this entry mirrors [this existing Codecademy blog post about Node.js](https://www.codecademy.com/article/what-is-node).

Dusch4593 replied:
I agree, though, that this is worded a little funny. I think I would rephrase as:

"Node allows developers to write server-side JS code that can directly access a computer's operating system and processes such as input/output and file storage."

Dusch4593 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,163 @@
+
+## Node Versions
+
+Node's major releases aim to support the latest JS features including ES6/ES2015 and beyond without transpilation. A resource called [node.green](http://node.green/) provides a list of supported JS features in different versions of Node.
```
Can do! I also think we should add a new General concept entry about compilers/transpilers so we can link back to a Docs entry in the future. 😄

Dusch4593 commented:
@HishamT Just finished implemented most of your feedback. Let me know if there's anything I missed. 😄

HishamT reviewed (approved):
Looks pretty good. I left some suggestions that you can implement if you wish.

Dusch4593 commented:
@HishamT Implemented your feedback. Let's get 1-2 more reviews for this PR before we merge 😄 

cc: @KTom101 @yangc95

KTom101 commented on content/ruby/concepts/ruby-on-rails/ruby-on-rails.md:
```diff
@@ -0,0 +1,57 @@
+## Installation
+
+Rails can be installed as a [Ruby gem](https://www.codecademy.com/resources/docs/ruby/gems). However, the following must be installed beforehand:
+
```
Since these need to be installed beforehand, do you think it would be wise to link out "SQLite3" and "Yarn"?

Dusch4593 replied:
Linked!

KTom101 reviewed (changes requested):
@Dusch4593 Added some suggestions mainly with linking out concepts.

Dusch4593 commented on content/general/concepts/hacking/hacking.md:
```diff
@@ -43,7 +44,7 @@ This puts system maintainers in a tough position as to whether it would be wise
+White hat hackers hack for the benefit of others, particularly organizations trying to discover (and close) security vulnerabilities in their system. They always obtain permission before penetrating an organization's system. Also known as "ethical hackers", many white hat hackers carry a Certified Ethical Hacker (CEH) qualification.
 
 A white hat hacker will normally engage directly with a company and listen to their concerns. Next, they will attempt to hack areas of their network to spot weaknesses. After they finish, they provide a report to the company and potential next steps.
 
```
Thank you for spotting that, @KTom101! 😄

Dusch4593 commented:
@KTom101 Thank you very much for looking over these files and leaving feedback! I've implemented the new changes along with a few extra edits including reducing the width of the Linux Torvalds image by half (currently, the image is gigantic and obscures the opening text upon first page load).

KTom101 reviewed (commented):
@Dusch4593 I have a few more comments.

yangc95 commented on content/general/concepts/back-end/back-end.md:
```diff
@@ -20,10 +20,9 @@ The back-end of a program or web application serves data to the [front-end](http
 
-- They place their order (or "request") with the waiter/waitress (or server).
+- They place their order (or "request") with the waiter/waitress (or [server](https://www.codecademy.com/resources/docs/general/server)).
 - The order is then taken to the kitchen (on the "back-end").
```
I agree with @KTom101, sounds a bit awkward. Maybe we can say "(representation of the back-end)" without the quotes around back-end. Also, can we do this for the contents within the parentheses above as well?

Dusch4593 commented on content/open-source/concepts/next-js/next-js.md:
```diff
@@ -0,0 +1,30 @@
+  - 'paths/computer-science'
+---
+
+[**Next.js**](https://nextjs.org) is an open-source development [framework](https://www.codecademy.com/resources/docs/general/framework) that brings the best of [React's](https://www.codecademy.com/resources/docs/react) client-side functionality to the [server-side](https://www.codecademy.com/resources/docs/general/server-side) when building web applicatons.
```
Yes, thanks for catching that. 😄

Dusch4593 commented:
Hey @KTom101 I believe I implemented all of your feedback in 9652fe7675a1830c167f32f04d44fdbd7bc6c858. I also deleted the `git.md` file under General since there is an entire Git topic already.

Dusch4593 commented:
@KTom101 Just realized this morning that the table never got fixed 😅 But it should be fixed now!

Dusch4593 commented:
@KTom101 Was there anything left to add, change, or check before merging?

yangc95 commented on content/open-source/concepts/graphql/graphql.md:
```diff
@@ -0,0 +1,38 @@
+
+## Why Adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with RESTful architecture. One of REST's disadvantages is it can result in an excessive number of [HTTP](https://www.codecademy.com/resources/docs/general/http) queries or too much data being fetched. Alternatively, GraphQL allows developers to ask for specific data instead of fetching a large object with all of the data every time. There i ...
```
- First sentence, should we use "adopting" rather than "discussing"
- Would it also be helpful to put these in bullet points to make each point clearly laid out?

yangc95 commented on content/open-source/concepts/graphql/graphql.md:
```diff
@@ -0,0 +1,38 @@
+
+## Who Is Using GraphQL?
+
+GraphQL is used by a wide range of businesses, from large to small. Some bigger companies include Airbnb, Lyft, [GitHub](https://www.codecademy.com/resources/docs/general/github), Twitter, Paypal, and Shopify. More information can be found [here](https://graphql.org/users/).
```
```suggestion
GraphQL is used by small and large businesses. Some bigger companies include Airbnb, Lyft, [GitHub](https://www.codecademy.com/resources/docs/general/github), Twitter, Paypal, and Shopify. More information can be found [here](https://graphql.org/users/).
```

yangc95 commented on content/open-source/concepts/next-js/next-js.md:
```diff
@@ -0,0 +1,30 @@
+
+- A page-based [routing](https://www.codecademy.com/resources/docs/general/routing) system that supports dynamic routes.
+- Every page is pre-rendered for improved site performance and search-engine optimization (SEO).
+- Any page can either be statically generated (SSG) ahead of time or rendered on the server-side (SSR) upon request.
```
SSG is static-site generator and SSR is server-side rendering

Is it okay to use these acronyms when these terms aren't actually being used?

Dusch4593 replied:
I see your point, these acronyms seem rather forced in the text. Let's omit them.

yangc95 commented on content/open-source/concepts/next-js/next-js.md:
```diff
@@ -0,0 +1,30 @@
+- A page-based [routing](https://www.codecademy.com/resources/docs/general/routing) system that supports dynamic routes.
+- Every page is pre-rendered for improved site performance and search-engine optimization (SEO).
+- Any page can either be statically generated (SSG) ahead of time or rendered on the server-side (SSR) upon request.
+- Faster page loads enables automatic code splitting or "lazy-loading".
```
```suggestion
- Faster page loads enables automatic code splitting or "lazy-loading" (load a resource only when it is needed).
```

yangc95 commented on content/open-source/concepts/next-js/next-js.md:
```diff
@@ -0,0 +1,30 @@
+
+Next.js was originally authored by Guillermo Rauch on [GitHub](https://www.codecademy.com/resources/docs/general/github) in 2016. Due to its functionality, large companies such as Netflix, Starbucks, Uber, GitHub, and Ticketmaster have all used Next.js for their website development. It is currently owned and maintained by the company [Vercel](https://vercel.com).
+
+**Fun Fact:** Codecademy Docs is built with Next.js.
```
This is nice! Are we able to do fun facts around Docs and if so, do we have some guidelines for it?

yangc95 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,165 @@
+
+Per its official home page, Node "uses an event-driven, non-blocking I/O model." In practice, this means that Node is built well to handle asynchronous JS code to perform many asynchronous activities such as reading and writing to the file system, handling connections to database servers, or handling requests as a web server.
+
+To handle asynchronous code, Node uses a callback-based system. Node functions and methods that will implement some asynchronous activity take a [callback function](https://www.codecademy.com/resources/docs/javascript/callbacks). This function will be called whenever the asynchronous operation has resolved. By convention, the first argument of this callback is an error placeholder. If an error oc ...
```
```suggestion
To handle asynchronous code, Node uses a callback-based system. Node functions and methods that will implement some asynchronous activity take a [callback function](https://www.codecademy.com/resources/docs/javascript/callbacks). This function will be called whenever the asynchronous operation has resolved. By convention, the first argument of this callback is an error placeholder. If an error in the asynchronous operation occurred (trying to read a non-existent file, for example), the error argument will be an `Error` object, that will be `null` if no error occurs.
```

yangc95 replied:
redundant here

yangc95 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,165 @@
+
+In this example, the user launches node on line one with the `node` terminal command.
+
+On line 2 the user types `4 + 5` and evaluates with the <kbd>Enter</kbd> key. `9` prints to the output terminal.
```
```suggestion
On line 2, the user types `4 + 5` and evaluates with the <kbd>Enter</kbd> key. `9` prints to the output terminal.
```

yangc95 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,165 @@
+
+After the script file is loaded, the variables are accessible in the REPL, so when we evaluate the `a` variable, it's value has been set by loading `script.js`, and 'Node REPL is fun!' prints to the console.
+
+This can be tested by running `node` in a terminal or check out the <a href="https://nodejs.org/api/repl.html" >REPL docs</a> for more functionality.
```
```suggestion
This can be tested by running `node` in a terminal or check out the <a href="https://nodejs.org/api/repl.html" >REPL docs</a> for more information on commands and functionality.
```

yangc95 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,165 @@
+
+## Package Management
+
+Node packages are a convenient way to share modules between Node developers. The service [npm](https://www.codecademy.com/resources/docs/javascript/npm) is the default package manager for Node; it becomes available after Node is installed.
```
```suggestion
Node packages are a convenient way to share modules between Node developers. The service Node Package Manager ([npm](https://www.codecademy.com/resources/docs/javascript/npm)) is the default package manager for Node; it becomes available after Node is installed.
```

yangc95 commented on content/open-source/concepts/node-js/node-js.md:
```diff
@@ -0,0 +1,165 @@
+
+### Version Management
+
+As with any major release of software, top-level Node releases (8.x, 7.x, etc.) sometimes introduce error-causing changes in applications built in older versions of the environment. A version manager can be used to switch between multiple versions of Node on a single computer.
```
```suggestion
As with any major release of software, top-level Node releases (8.x, 7.x, etc.) sometimes introduce error-causing changes in applications built using older versions on an environment. A version manager can be used to switch between multiple versions of Node on a single computer.
```

yangc95 commented on content/open-source/open-source.md:
```diff
@@ -0,0 +1,13 @@
+---
+Title: 'Open Source' # This is the name that appears on the hub page for this language. Pay attention to capitalization and punctuation!
```
```suggestion
Title: 'Open Source'
```

yangc95 reviewed (commented):
@Dusch4593 I had a few more comments/suggestions! TY for making all these changes and additions!

Dusch4593 commented on content/ruby/concepts/ruby-on-rails/ruby-on-rails.md:
````diff
@@ -0,0 +1,57 @@
+gem install rails
+```
+
+## The MVC Pattern
````
I don't see why not 😄

Dusch4593 commented:
@yangc95 Changes implemented 😄

KTom101 reviewed (approved):
@Dusch4593 Looks good to me!

yangc95 reviewed (approved):
LGTM as well~

Dusch4593 commented:
@SSwiniarski @yangc95 @KTom101 @HishamT Thank you so much to everyone for their input and feedback for this new topic. This one was a long time coming. 😄 Live link for the new entries below:

Open Source (topic): https://www.codecademy.com/resources/docs/open-source
Open Source Software (concept): https://www.codecademy.com/resources/docs/open-source/open-source-software
